cinema/views.py

```diff
@@ -24,21 +27,29 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication, )
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly, )
```

Comment:
The `GenreViewSet` should only allow list and create actions according to the task requirements. Consider overriding the `get_permissions` method to restrict actions accordingly.